### ISSUE TITLE
Google Analytics Capstone Project

### ISSUE THREAD

**Issue by abhisheks008** [abhisheks008/ML-Crate#474]
### ML-Crate Repository (Proposing new issue)
:red_circle: **Project Title** : Google Analytics Capstone Project
:red_circle: **Aim** : Create an analysis model for the Google analytics using machine learning.
:red_circle: **Dataset** : https://www.kaggle.com/datasets/fredericxiong/google-analytics-capstone-project
:red_circle: **Approach** : Try to use 3-4 algorithms to implement the models and compare all the algorithms to find out the best fitted algorithm for the model by checking the accuracy scores. Also do not forget to do a exploratory data analysis before creating any model.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : 
- GitHub Profile Link : 
- Participant ID (If not, then put NA) :
- Approach for this Project :
- What is your participant role? (Mention the Open Source Program name. Eg. HRSoC, GSSoC, GSOC etc.)

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by abhisheks008** [comment]
Issue assigned to you @Shrutakeerti

**Comment by abhisheks008** [comment]
Unassigned as the open source event ended up.

**Comment by why-aditi** [comment]
Full name : Aditi Kala
GitHub Profile Link : https://github.com/why-aditi
Participant ID (If not, then put NA) : N/A
Approach for this Project :
Data Collection and Preparation -> EDA -> Model Training -> Model Validation -> Comparing the performance metrics of various models
What is your participant role? SSOC'24

**Comment by abhisheks008** [comment]
> Full name : Aditi Kala GitHub Profile Link : https://github.com/why-aditi Participant ID (If not, then put NA) : N/A Approach for this Project : Data Collection and Preparation -> EDA -> Model Training -> Model Validation -> Comparing the performance metrics of various models What is your participant role? SSOC'24

Can you share a brief about the planned models?

**Comment by fspzar123** [comment]
Full name : Filbert Shawn
GitHub Profile Link : https://github.com/fspzar123
Participant ID (If not, then put NA) : N/A
Approach for this project: 
**Initial processing:**
- Address any values that are missing.
- If required, standardize or normalize the data.

**Illustration:**
- Plot each region's time series data to identify trends and patterns.
- To see how several regions relate to one another, use correlation charts.
- Model Selection: Exponential Smoothing, XGBoost, LSTM, and ARIMA.
- Divide the data into sets for testing and training.
- Utilize various models and assess each one's performance with metrics such as MAE, RMSE, and MAPE.
- Cross-validation should be done to make sure the model is robust.

**Adjusting Hyperparameters:**
- Make use of grid search or random search to fine-tune the parameters of the models I have selected.

**Predicting:**
- Utilize the trained model to predict future values and compare them with the test set.

**Results visualization:**
- Plot the actual numbers against the projected values to see how well your model is working.

What is your participant role? SSOC 24

**Comment by why-aditi** [comment]
> > Full name : Aditi Kala GitHub Profile Link : https://github.com/why-aditi Participant ID (If not, then put NA) : N/A Approach for this Project : Data Collection and Preparation -> EDA -> Model Training -> Model Validation -> Comparing the performance metrics of various models What is your participant role? SSOC'24
> 
> Can you share a brief about the planned models?

I'm thinking of using Random Forest, ARIMA, LSTM at the moment

**Comment by abhisheks008** [comment]
> > > Full name : Aditi Kala GitHub Profile Link : https://github.com/why-aditi Participant ID (If not, then put NA) : N/A Approach for this Project : Data Collection and Preparation -> EDA -> Model Training -> Model Validation -> Comparing the performance metrics of various models What is your participant role? SSOC'24
> > 
> > 
> > Can you share a brief about the planned models?
> 
> I'm thinking of using Random Forest, ARIMA, LSTM at the moment

Cool then go ahead. Try to implement 5-6 models for this problem statement.

Assigned this issue to you.
